tests/testcontainers/test_h2spec.rs

```diff
@@ -225,8 +288,69 @@ async fn test_h2spec() {
     assert_eq!(0, router_exit_code.unwrap());
 }
 
+struct LogPrinter {
+    tasks: Vec<JoinHandle<()>>,
+}
+
+impl Drop for LogPrinter {
+    fn drop(&mut self) {
+        for task in self.tasks.drain(..).rev() {
+            let result = futures::executor::block_on(task);
+            println!("Finalized LogPrinter {:?}", result);
+        }
+    }
+}
+
+impl LogPrinter {
+    fn new() -> Self {
+        Self {
+            tasks: vec![],
+        }
+    }
+
+    fn print(&mut self, log_stream: Fuse<(impl Stream<Item=String> + Unpin + Sized + Send + 'static)>) {
```

Comment:
yes, the type signature is kind of crazy, and even IDE plugin has some trouble with it, turns out

* https://github.com/intellij-rust/intellij-rust/issues/10363